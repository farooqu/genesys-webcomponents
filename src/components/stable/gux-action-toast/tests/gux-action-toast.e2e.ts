import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

describe('gux-action-toast', () => {
  describe('#render', () => {
    [
      {
        description: 'should render action toast',
        html: `
          <gux-action-toast lang="en" accent="neutral">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
            <gux-button slot="negative-button">Reject</gux-button>
            <gux-button slot="positive-button" accent="primary">Accept</gux-button>
          </gux-action-toast>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-action-toast');
        await a11yCheck(page);
        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
