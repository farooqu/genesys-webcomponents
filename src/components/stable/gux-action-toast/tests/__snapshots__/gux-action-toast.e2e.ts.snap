// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-action-toast #render should render action toast 1`] = `"<gux-action-toast lang=\\"en\\" accent=\\"neutral\\" class=\\"hydrated\\"> <gux-icon slot=\\"icon\\" icon-name=\\"user-add\\" decorative=\\"\\" class=\\"hydrated\\"></gux-icon> <div slot=\\"title\\">Title</div> <div slot=\\"message\\">This is the message</div> <gux-button slot=\\"negative-button\\" class=\\"hydrated\\">Reject</gux-button> <gux-button slot=\\"positive-button\\" accent=\\"primary\\" class=\\"hydrated\\">Accept</gux-button> </gux-action-toast>"`;
